### ISSUE TITLE
Report unnecessary case statements in match statement

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4689]
If a `case` pattern can never be matched, it would be useful to report this, as it's likely the sign of a bug.

See https://github.com/microsoft/pylance-release/issues/4005.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.296, which I just published. It will also be included in a future release of pylance.

**Comment by AdrianB-sovo** [comment]
@erictraut Except that we may do something like that:
```python
match Language:
    case Language.English:
        # Something...
    case Language.French:
        # Something...
    case _:
        # New language was added, but we forgot to handle it here.
        raise Exception("Language not supported")
```

The ideal things would probably to not have this error if an exception is raised or an assert is done in the wildcard case, but I have no idea if it's feasible or worth it.
